config/types/etcd.go

```diff
@@ -147,6 +150,47 @@ func etcdContents(etcd Etcd, platform string) (string, error) {
 	return unit.String(), nil
 }
 
+var validPrefixes = []string{"http://", "https://", "unix://", "unixs://"}
+
+func checkForPrefix(s string) bool {
+	if s == "" {
+		return true
+	}
+	urlStrs := strings.Split(s, ",")
+	for _, urlStr := range urlStrs {
+		hasPrefix := false
+		for _, p := range validPrefixes {
+			hasPrefix = hasPrefix || strings.HasPrefix(urlStr, p)
+		}
+		if !hasPrefix {
+			return false
+		}
+	}
+	return true
+}
+
+func validatePrefixes(m map[string]string) report.Report {
+	r := report.Report{}
+	for name, val := range m {
+		if !checkForPrefix(val) {
+			r.Add(report.Entry{
+				Message: fmt.Sprintf("%s doesn't start with one of: %v", name, validPrefixes),
```

Comment:
The field name will never start with one of those prefixes. "value for %s doesn't..."